d3rlpy/base.py

```diff
@@ -423,7 +442,7 @@ def _evaluate(self, episodes, scorers, logger):
             logger.add_metric(name, test_score)
 
             # store metric locally
-            if name not in self.eval_results_:
+            if name not in self.eval_results_ and test_score is not None:
                 self.eval_results_[name] = []
```

Comment:
You can remove this if you use `defaultdict`.

Reply:
exactly !